nixos/modules/phoebus/save-and-restore.nix

```diff
@@ -0,0 +1,155 @@
+{
+  config,
+  epnixLib,
+  lib,
+  pkgs,
+  ...
+}: let
+  cfg = config.services.phoebus-save-and-restore;
+  settingsFormat = pkgs.formats.javaProperties {};
+  configFile = settingsFormat.generate "phoebus-save-and-restore.properties" cfg.settings;
+
+  localElasticsearch = cfg.settings."elasticsearch.network.host" == "localhost";
+in {
+  options.services.phoebus-save-and-restore = {
+    enable = lib.mkEnableOption ''
+      the Phoebus Save-and-restore service.
+
+      This service is used by clients
+      to manage configurations (aka save sets) and snapshots,
+      to compare snapshots,
+      and to restore PV values from snapshots.
+    '';
+
+    openFirewall = lib.mkOption {
+      description = ''
+        Open the firewall for the Phoebus Save-and-restore service.
+
+        Warning: this opens the firewall on all network interfaces.
+      '';
+      type = lib.types.bool;
+      default = false;
+    };
+
+    settings = lib.mkOption {
+      description = ''
+        Configuration for the Phoebus Alarm Logger.
+
+        These options will be put into a `.properties` file.
+
+        Note that options containing a "." must be quoted.
+
+        Available options can be seen here:
+        <https://github.com/ControlSystemStudio/phoebus/blob/master/services/alarm-logger/src/main/resources/application.properties>
+      '';
+      default = {};
+      type = lib.types.submodule {
+        freeformType = settingsFormat.type;
+        options = {
+          "server.port" = lib.mkOption {
+            description = "Port for the Alarm Logger service";
+            type = lib.types.port;
+            default = 8080;
```

Comment:
Use default port of the official document instead

Reply:
In the end, 8080 is still documented as the default port. I've opened https://github.com/epics-extensions/EPNix/issues/37 to discuss this further.